include/zephyr/posix/unistd.h

```diff
@@ -117,7 +117,7 @@ extern "C" {
 #define _XOPEN_REALTIME			  (-1L)
 #define _XOPEN_REALTIME_THREADS		  (-1L)
 #define _XOPEN_SHM			  (-1L)
-#define _XOPEN_STREAMS			  (-1L)
+#define _XOPEN_STREAMS			   _POSIX_VERSION
```

Comment:
I thought we had a Kconfig option for this already, but if not, then now would be the correct time to make one.
```suggestion
#define _XOPEN_STREAMS Z_SC_VAL_IFDEF(CONFIG_XOPEN_STREAMS, _POSIX_VERSION)
```

Reply:
Aha - actually, we have `CONFIG_POSIX_PUTMSG`. Maybe it would make sense to change that to `CONFIG_XOPEN_STREAMS` instead.

Reply:
Could be better indeed.